File: algorithms/338. Counting Bits/solution.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        count = [0] * (num + 1)
        for i in xrange(1, num + 1):
            count[i] = count[i >> 1] + (i & 1)
        return count

File: algorithms/338. Counting Bits/solution.cc
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> count(num + 1);
        for ( int i = 0; i <= num; ++i ) {
            count[i] = count[i >> 1] + (i & 1);
        }
        return count;
    }
};
